ga4gh/backend.py

```diff
@@ -184,30 +182,31 @@ class AbstractBackend(object):
     This class provides methods for all of the GA4GH protocol end points.
     """
     def __init__(self):
-        self._variantSetIdMap = {}
-        self._variantSetIds = []
         self._referenceSetIdMap = {}
         self._referenceSetIds = []
-        self._readGroupSetIdMap = {}
-        self._readGroupSetIds = []
-        self._readGroupIds = []
-        self._readGroupIdMap = {}
         self._requestValidation = False
         self._responseValidation = False
         self._defaultPageSize = 100
         self._maxResponseLength = 2**20  # 1 MiB
+        self._dataset = datasets.AbstractDataset()
+
+    def getDataset(self):
+        """
+        Returns the backend's dataset
+        """
+        return self._dataset
 
     def getVariantSets(self):
         """
         Returns the list of VariantSets in this backend.
         """
-        return list(self._variantSetIdMap.values())
+        return list(self.getDataset().getVariantSetIdMap().values())
 
     def getReadGroupSets(self):
         """
         Returns the list of ReadGroupSets in this backend.
         """
-        return list(self._readGroupSetIdMap.values())
+        return list(self.getDataset().getReadGroupSetIdMap().values())
```

Comment:
These methods are only used by the frontend code for displaying the dataset summaries. We should just move the `getReadGroupSets` and `getVariantSets` methods into DataSet, and have the frontend code work directly with the dataset object.